.ci/test_r_package.sh

```diff
@@ -20,6 +20,12 @@ export _R_CHECK_CRAN_INCOMING_REMOTE_=0
 # to catch extreme problems
 export _R_CHECK_PKG_SIZES_THRESHOLD_=60
 
+# don't fail builds for long-running examples unless they're very long
+# See https://github.com/microsoft/LightGBM/issues/4049#issuecomment-793412254.
+if [[ $R_BUILD_TYPE != "cran" ]]; then
```

Comment:
In case you are wondering "why only do this on non-cran builds?", my proposal from https://github.com/microsoft/LightGBM/issues/4049#issuecomment-793412254

> I also think it would be a good idea to only do that when BUILD_TYPE = "cmake", just in case the behavior of R CMD check --as-cran changes in the future (like how it started running \donttest{} examples automatically in R 4.0.0, https://cran.r-project.org/doc/manuals/r-devel/NEWS.html).